web/pages/reports.tsx

```diff
@@ -0,0 +1,297 @@
+) => {
+  const response = await api('update-report', { reportId, updates })
+  if (response.status === 'success') {
+    return response.data
+  } else {
+    console.error('Error updating report:', response)
+    return null
+  }
+}
+
+const App = () => {
+  const isAdminOrMod = useAdminOrMod()
+  const [reports, setReports] = useState<Report[]>([])
+  const [initialLoading, setInitialLoading] = useState(true)
+  const [owner, setOwner] = useState<{ [key: string]: any }>({})
+  const [reportStatuses, setReportStatuses] = useState<{
+    [key: number]: ReportStatus
+  }>({})
+  const [modNotes, setModNotes] = useState<{
+    [key: number]: string | undefined
+  }>({})
+  const initialLoadRef = useRef(true)
+
+  const getReports = async () => {
+    const response = await api('get-reports', {})
+    if (response && response.status === 'success') {
+      const newReports = response.data
+
+      if (JSON.stringify(newReports) !== JSON.stringify(reports)) {
+        const sortedReports = newReports
+          .filter((report: Report) => report.created_time)
+          .sort(
+            (a: Report, b: Report) =>
+              new Date(b.created_time ?? 0).getTime() -
+              new Date(a.created_time ?? 0).getTime()
+          )
+
+        setReports(sortedReports)
+        const initialStatuses = sortedReports.reduce(
+          (acc: any, report: Report) => {
+            acc[report.report_id] = report.status
+            return acc
+          },
+          {} as { [key: number]: ReportStatus }
+        )
+        const initialNotes = sortedReports.reduce(
+          (acc: any, report: Report) => {
+            acc[report.report_id] = report.mod_note
+            return acc
+          },
+          {} as { [key: number]: string | undefined }
+        )
+        setReportStatuses(initialStatuses)
+        setModNotes(initialNotes)
+      }
+    } else {
+      console.error('Failed to fetch reports:', response)
+    }
+
+    if (initialLoadRef.current) {
+      setInitialLoading(false)
+      initialLoadRef.current = false
+    }
+  }
+
+  useEffect(() => {
+    getReports()
+
+    const intervalId = setInterval(() => {
+      getReports()
+    }, 1000)
+
+    return () => clearInterval(intervalId)
+  }, [])
+
+  const handleStatusChange = async (
+    reportId: number,
+    newStatus: ReportStatus
+  ) => {
+    setReportStatuses((prevStatuses) => ({
+      ...prevStatuses,
+      [reportId]: newStatus,
+    }))
+
+    await updateReport(reportId, { status: newStatus })
+  }
+
+  const handleNoteSave = async (reportId: number, newNote: string) => {
+    setModNotes((prevNotes) => ({
+      ...prevNotes,
+      [reportId]: newNote,
+    }))
+
+    await updateReport(reportId, { mod_note: newNote })
+  }
+
+  useEffect(() => {
+    const fetchOwner = async () => {
+      const ownerData: { [key: string]: any } = {}
+      for (const report of reports) {
+        const user = await getUserById(report.user_id)
```

Comment:
If you need the User object, you should return that in the endpoint instead of the user_id. Does that make sense?

I.e. query for the user within the endpoint, not on the client.